stacks-signer/src/v0/signer.rs

```diff
@@ -587,7 +589,20 @@ impl Signer {
             block_info.signed_self.get_or_insert(get_epoch_time_secs());
         }
         // Record the block validation time
-        block_info.validation_time_ms = Some(block_validate_ok.validation_time_ms);
+        let non_bootcode_contract_call_block = block_info.block.txs.iter().any(|tx| {
```

Comment:
Instead of iterating the transactions, can you just check the cost in the `block_validate_ok` and if it is all 0s, then set the validation time to 0.

Reply:
You mean eagerly check the validation time to prevent needless iteration? I think the only time validation_time_ms would be 0 is if there are no txs to iterate over? so what is the benefit?

Reply:
I think he means that, because you get `ExecutionCost` in `BlockValidateOk`, you can just set `validation_time_ms` to 0. So like (pseudocode):

```rust
if validation.execution_cost.is_zero() { 0 } else { validation.validation_time_ms }
```

Reply:
Right. A block with only STX transfers has 0 cost.

Reply:
OOOOO I am dumb. Yes of course this makes sense. I was thinking you were saying validation.validation_time_ms check if its 0...and I didn't understand why we would do the transfer tx check at all then. Now it makes sense.

Reply:
Done as part of https://github.com/stacks-network/stacks-core/pull/5494